plugins/org.eclipse.fordiac.ide.application/src/org/eclipse/fordiac/ide/application/commands/MoveElementsFromSubAppCommand.java

```diff
@@ -61,23 +54,28 @@ public class MoveElementsFromSubAppCommand extends Command implements ScopedComm
 	private Position destination;
 	private final FBNetwork destinationNetwork;
 	protected final List<FBNetworkElement> elements;
-	private final Map<FBNetworkElement, org.eclipse.fordiac.ide.model.libraryElement.Position> oldPos = new HashMap<>(); // for
-	// undo
-	private final Map<FBNetworkElement, org.eclipse.fordiac.ide.model.libraryElement.Position> newPos = new HashMap<>(); // for
-	// redo
+	private final Map<FBNetworkElement, Position> oldPos = new HashMap<>(); // for undo
+	private final Map<FBNetworkElement, Position> newPos = new HashMap<>(); // for redo
 	private final CompoundCommand unmappingCmds = new CompoundCommand(); // stores all needed unmap commands
-	private final CompoundCommand deleteConnectionsAndInterfaceElements = new CompoundCommand();
-	private final CompoundCommand createConnectionsCommands = new CompoundCommand();
-	protected final CompoundCommand createSubAppInterfaceElementCommands = new CompoundCommand();
 	private final CompoundCommand setUniqueName = new CompoundCommand();
 	private final CompoundCommand removeFromGroup = new CompoundCommand();
 	private final Set<Connection> connsMovedToParent = new HashSet<>();
+	private CompoundCommand reconnectConnectionsCommands = null;
 
 	public MoveElementsFromSubAppCommand(final Collection<FBNetworkElement> elements, final Point destination) {
+		this(elements, destination, null);
+	}
+
+	public MoveElementsFromSubAppCommand(final Collection<FBNetworkElement> elements, final Point destination,
+			final FBNetwork destinationNetwork) {
 		this.elements = new ArrayList<>(elements);
 		setDestination(destination);
 		this.sourceSubApp = getSourceSubapp();
-		this.destinationNetwork = sourceSubApp != null ? sourceSubApp.getFbNetwork() : null;
+		if (destinationNetwork == null) {
+			this.destinationNetwork = sourceSubApp != null ? sourceSubApp.getFbNetwork() : null;
+		} else {
+			this.destinationNetwork = destinationNetwork;
+		}
 	}
 
 	private SubApp getSourceSubapp() {
```

Comment:
what happens when the element to be moved is inside the application fbNetwork ?

Reply:
This has now been added